src/main/java/org/spongepowered/common/mixin/core/block/MixinBlockPistonBase.java

```diff
@@ -75,25 +73,16 @@
      * @param direction The direction to move in
      * @param extending Whether the piston is in an extended state
      */
-    @Inject(method = "doMove", at = @At("HEAD"), cancellable = true)
+    @Inject(method = "doMove", at = @At("HEAD"))
     public void onDoMoveHead(World worldIn, BlockPos pos, EnumFacing direction, boolean extending, CallbackInfoReturnable<Boolean> ci) {
         if (!worldIn.isRemote && CauseTracker.ENABLED) {
             final CauseTracker causeTracker = ((IMixinWorldServer) worldIn).getCauseTracker();
-            final PhaseData data = causeTracker.getCurrentPhaseData();
-            BlockSnapshot source = data.context.getSource(BlockSnapshot.class).orElse(null);
-            BlockPos targetPos = pos.offset(direction);
-            final IBlockState blockstate = worldIn.getBlockState(pos);
-            if (source == null) {
-                source = ((IMixinWorldServer) worldIn).createSpongeBlockSnapshot(blockstate, blockstate, pos, 2);
-            }
-            if (!worldIn.isRemote && SpongeCommonEventFactory.handleChangeBlockEventPre((IMixinWorldServer) worldIn, targetPos, source)) {
-                ci.setReturnValue(false);
-                return;
-            }
 
+            final IBlockState currentState = worldIn.getBlockState(pos);
+            final SpongeBlockSnapshot snapshot = ((IMixinWorldServer) worldIn).createSpongeBlockSnapshot(currentState, currentState, pos, 0);
```

Comment:
The original source was intended to be pre-captured of the tile entity before the move event happens, or, creating it if it wasn't already created.